library/std/src/sys/pal/unix/mod.rs

```diff
@@ -227,6 +244,14 @@ pub(crate) fn unix_sigpipe_attr_specified() -> bool {
     UNIX_SIGPIPE_ATTR_SPECIFIED.load(crate::sync::atomic::Ordering::Relaxed)
 }
 
+#[cfg(not(any(
+    target_os = "espidf",
+    target_os = "emscripten",
+    target_os = "fuchsia",
+    target_os = "horizon",
+)))]
+extern "C" fn _rustc_sigaction_noop(_: libc::c_int) {}
```

Comment:
Maybe put this next to the `sa_sigaction` assignment? Or do `new.sa_sigaction = |_: lib::c_init| as libc::sighandler_t;` instead?